docs/content/tutorial/managing-your-own-io.mdx

```diff
@@ -0,0 +1,106 @@
+- You want to run a SQL query that creates or updates a table on the database.
+- Adjacent to the above point, your I/O manager is batch-oriented, and the data loaded would not fit into memory.
+- Writing your pipeline using I/O managers is possible, but you don't want use this Dagster-specific concept. For example, you have an existing pipeline and don't want to refactor it.
+
+By the end of this section, you'll also have assets that depend on other assets without loading the dependent asset's data into memory.
+
+## Step 1: Creating an asset without an I/O Manager
+
+Many valid situations exist where you would want to avoid using an I/O manager to write an asset back to memory. One of these situations is downloading a file from the internet. Because the file is downloaded directly onto the machine rather than returning it in memory, there is no need to use an I/O manager to write the data back to storage.
+
+In your `assets.py`, create a new asset called `stopwords_zip` by copying and pasting the code below:
+
+```python file=/tutorial/managing_your_own_io/assets.py startafter=start_stopwords_zip endbefore=end_stopwords_zip
+import urllib.request  # Addition: new import at the top of `assets.py`
+
+from dagster import asset
+
+
+@asset
+def stopwords_zip() -> None:
+    urllib.request.urlretrieve(
+        "https://docs.dagster.io/assets/stopwords.zip,data/stopwords.zip",
+    )
+```
+
+In this asset, you’ll download a file from the internet. Dagster stores this file on the machine it is running on. There are two key differences between the assets you’ve written so far and this asset:
+
+- **It doesn't have a return statement.** Because the zip file is downloaded directly onto the machine, there is no need to return data.
+- **It has a `None` annotation for its return type.** This tells Dagster that it shouldn’t use an I/O manager because there is no data to write back.
+
+These are the only two changes required to have an asset not use an I/O manager.
+
+Refresh your code location, observe the new asset, and materialize it as you have with the previous assets. You can confirm that this works by finding the `stopwords.zip` in the `/data` directory made in the **Saving your data** section of the tutorial.
+
+## Step 2: Depend on an asset without loading it
+
+You’ll now unzip the zip file asset you made and extract the CSV from it. This CSV will be a new asset that depends on the zip file asset you downloaded.
+
+By default, an asset will load its dependencies into memory when it materializes. However, there are situations where you would want to avoid doing this. If you're making changes directly on data in storage, such as unzipping a zip file, you don't need to load the data into memory.
+
+To tell an asset not to load a dependency into memory when materializing, you can use the `@asset` decorator’s `non_argument_deps` parameter to establish the dependency instead of passing it in as an argument of the asset’s function definition.
```

Comment:
```suggestion
To tell an asset not to load a dependency into memory when materializing, you can use the `@asset` decorator’s `non_argument_deps` parameter to establish the dependency, instead of including it as an argument of the asset’s function definition.
```